WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReaders/CardReaderSettingsViewModel.swift

```diff
@@ -162,10 +162,19 @@ final class CardReaderSettingsViewModel: ObservableObject {
 
     /// Views can call this method to disconnect from the connected reader.
     func disconnectAndForget() {
-        // TODO dispatch an action to disconnect.
-        connectedReader = nil
-        activeView = .connectYourReader
-        activeAlert = .none
+        let action = CardPresentPaymentAction.disconnect { [weak self] result in
+            guard let self = self else {
+                return
+            }
+
+            if result.isSuccess {
```

Comment:
Not super important, but maybe that condition could also go in the `guard` above